x-pack/legacy/plugins/monitoring/public/np_ready/plugin.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  CoreSetup,
+  CoreStart,
+  LegacyCoreStart,
+  Plugin,
+  PluginInitializerContext
+} from 'kibana/public';
+
+/**
+ * TODO: pass: uiRoutes, uiModules, and configureAppAngularModule (or their alternatives) outside np_ready
+ */
+import uiRoutes from 'ui/routes';
+//@ts-ignore
+import { uiModules } from 'ui/modules';
+import { configureAppAngularModule } from 'ui/legacy_compat';
+
+
+//@ts-ignore
+import { localApplicationService } from 'plugins/kibana/local_application_service';
+
+export class MonitoringPlugin implements Plugin {
+  constructor(ctx: PluginInitializerContext) { }
+
+  public setup(core: CoreSetup, dep: any) {
+    //TODO
+  }
+
+  public start(core: CoreStart) {
+    const kibanaModule = uiModules.get('kibana');
+    configureAppAngularModule(kibanaModule, core as LegacyCoreStart, true);
+
+    core.uiSettings.set('timepicker:timeDefaults', JSON.stringify({
```

Comment:
is it okay that plugin changes the global settings?

Reply:
This is an alternative to `chrome.getUiSettingsClient().overrideLocalDefault()`, right?

I guess I could just pass those setting directly to the timefilter without overwriting anything

Reply:
> I guess I could just pass those setting directly to the timefilter without overwriting anything

sounds good then